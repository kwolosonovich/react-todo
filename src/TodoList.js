import React, {useState} from "react";
import Item from './Item'
import NewItemForm from "./NewItemForm";

const TodoList = () => {
    const INITIAL_STATE = [
      { id: 1, name: "Shopping", date: "12/1/20" }
    ]
    const [items, setItems] = useState(INITIAL_STATE);
    const addItem = (name, date) => {
      setItems(items => [...items, {name, date}])
    }
    return (
      <div>
        <h3>To-Do List</h3>
        <NewItemForm />
        <div>
          {items.map(({id, name, date}) => <Item id={id} name={name} key={id} date={date} />)}
        </div>
      </div>
    );
}

export default TodoList;