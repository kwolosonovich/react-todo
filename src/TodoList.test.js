import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";


test("renders without crashing", () => {
  render(<TodoList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

test("add new todo item", () => {
  const {queryByText, getByLabelText} = render(<TodoList />)
  const input = getByLabelText("To-Do")
  const btn = queryByText('Add');
  expect(queryByText('Baking')).not.toBeInTheDocument();
  fireEvent.change(input, {target: {name: 'Baking'}})
  firstEvent.click(btn);
  expect(queryByText('To-Do: Baking')).toBeInTheDocument();
})

