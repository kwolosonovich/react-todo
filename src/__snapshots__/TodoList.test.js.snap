// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      To-Do List
    </h3>
    <div>
      <form>
        <label
          for="name"
        >
          To-Do: 
        </label>
        <input
          id="name"
          name="name"
          placeholder="Add"
          type="text"
          value=""
        />
        <label
          for="date"
        >
          Due By: 
        </label>
        <input
          id="date"
          name="date"
          placeholder="Date"
          type="text"
          value=""
        />
        <button>
          Add
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
